### ISSUE TITLE
`no-useless-computed-key` should check class members

### ISSUE THREAD

**Issue by ark120202** [eslint/eslint#12041]
**What rule do you want to change?**

[no-useless-computed-key](https://eslint.org/docs/rules/no-useless-computed-key)

**Does this change cause the rule to produce more or fewer warnings?**

More

**How will the change be implemented? (New option, new default behavior, etc.)?**

I think it should be new default behavior, but if breaking changes should be avoided it could be a new option that would default to `true` (or removed) in the next major release.

**Please provide some example code that this change will affect:**

```js
class Foo {
	["bar"]() {}
}
```

**What does the rule currently do for this code?**

No warnings

**What will the rule do after it's changed?**

Produce a warning at `["bar"]() {}`.

**Are you willing to submit a pull request to implement this change?**

Yes

**Comment by platinumazure** [comment]
I would support this change. Let's implement it as an option to avoid the breaking change, and change the default in the next major release.

**Comment by platinumazure** [comment]
@eslint/eslint-team I'll champion this. Can we get one more :+1: here?

**Comment by mdjermanovic** [comment]
:thumbsup:
